utils/zerovec/src/ule/plain.rs

```diff
@@ -199,3 +199,73 @@ impl AsULE for i8 {
 
 // EqULE is true because i8 is its own ULE.
 unsafe impl EqULE for i8 {}
+
+// These impls are actually safe and portable due to Rust always using IEEE 754, see the documentation
+// on f32::from_bits: https://doc.rust-lang.org/stable/std/primitive.f32.html#method.from_bits
+//
+// The only potential problem is that some older platforms treat signaling NaNs differently. This is
+// still quite portable, signalingness is not typically super important.
+
+impl AsULE for f32 {
```

Comment:
Consider making these also be `EqULE`, because the bit pattern of `&[f32]` equals the bit pattern of `&[f32::ULE]` according to the docs page linked above on all current platforms. Or, don't do this if you think that situation might change.